profile/src/test/java/com/scottlogic/deg/profile/reader/JsonProfileReaderTests.java

```diff
@@ -44,7 +45,7 @@
     JsonProfileReader jsonProfileReader = new JsonProfileReader(
         null,
         new MainConstraintReader(
-            new AtomicConstraintTypeReaderMap(null)));
+            new AtomicConstraintTypeReaderMap(), new AtomicConstraintValueReader(null)));
```

Comment:
not big fan of passing nulls around as feels like we're breaking the contract of the class testing and its confusing and implies we might have to be careful way we call the methods on the class within this test to make sure we don't get a null ptr exception.  This decreases its maintainability as a test